pkg/ccl/crosscluster/logical/logical_replication_job_test.go

```diff
@@ -2273,4 +2273,18 @@ func TestLogicalReplicationCreationChecks(t *testing.T) {
 	dbA.Exec(t, "CANCEL JOB $1", jobAID)
 	jobutils.WaitForJobToCancel(t, dbA, jobAID)
 	replicationtestutils.WaitForAllProducerJobsToFail(t, dbB)
+
+	// Add different default values to to the source and dest and verify the
+	// stream can be created.
+	dbA.Exec(t, "ALTER TABLE tab ADD COLUMN new_col_2 INT NOT NULL DEFAULT 10")
+	dbB.Exec(t, "ALTER TABLE tab ADD COLUMN new_col_2 INT NOT NULL DEFAULT 20")
```

Comment:
I think it'd be good to insert on both sides with replication running and observe the insert-populated default just to prove how it is working.

That said, I think the really interesting cases arise if you have fewer columns in the src than in the dest and are expected to eval the defaults.